src/Action/AbstractPostArchiveAction.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\NewsBundle\Action;
+
+use Sonata\NewsBundle\Model\BlogInterface;
+use Sonata\NewsBundle\Model\PostManagerInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+abstract class AbstractPostArchiveAction extends Controller
+{
+    /**
+     * @var BlogInterface
+     */
+    private $blog;
+
+    /**
+     * @var PostManagerInterface
+     */
+    private $postManager;
+
+    public function __construct(BlogInterface $blog, PostManagerInterface $postManager)
+    {
+        $this->blog = $blog;
+        $this->postManager = $postManager;
+    }
+
+    /**
+     * NEXT_MAJOR: make this method protected.
+     *
+     * @return Response
+     */
+    final public function renderArchive(Request $request, array $criteria = [], array $parameters = [])
+    {
+        $pager = $this->postManager->getPager(
+            $criteria,
+            $request->get('page', 1)
+        );
+
+        $parameters = array_merge([
+            'pager' => $pager,
+            'blog' => $this->blog,
+            'tag' => false,
+            'collection' => false,
+            'route' => $request->get('_route'),
+            'route_parameters' => $request->get('_route_params'),
+        ], $parameters);
+
+        $response = $this->render(
```

Comment:
Too soon. We still support symfony 2.8: https://github.com/symfony/symfony/blob/2.8/src/Symfony/Bundle/FrameworkBundle/Controller/Controller.php#L184-L190

Reply:
@greg0ire PR about controller actions and PR about decoupling from templating are not related to each other (my PR was about using templating service inside blocks). So, i think, if you would inject twig into controller action, it will work. But, may be some other actions use other Controller/ControllerTrait features?

Reply:
Maybe yeah, but there's probably not that many, and if it can spare us from injecting the whole container, then it might be good. I might try it on the new actions I created but technically it will be a BC-break.